Chapter08/recipe-06/cxx-example/CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+cmake_minimum_required(VERSION 3.5 FATAL_ERROR)
+
+project(recipe-06 CXX)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+include(GNUInstallDirs)
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR})
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR})
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR})
+
+# defines targets and sources
+include(src/CMakeLists.txt)
```

Comment:
`add_subdirectory(src)` doesn't cut it here?

Reply:
In this case it would work and be better but I wanted to go "all in" with include and discuss when add_subdirectory would not work. Let's see perhaps I place add_subdirectory here and discuss this in the text.